### ISSUE TITLE
[k8s] Make `sky local up` support GPU

### ISSUE THREAD

**Issue by concretevitamin** [skypilot-org/skypilot#2889]
A common request from user is they have a local GPU node, and they'd like to use SkyPilot to run jobs on it.

Currently, `sky local up` uses `kind` which doesn't natively support GPUs. We should investigate workarounds (e.g., is labeling the GPU node in the kind k8s cluster good enough?).

-----
I tried `sky local up` which works on a L4 node, but cannot detect or label the GPUs:
```
$ python -m sky.utils.kubernetes.gpu_labeler
Found 0 GPU nodes in the cluster
No GPU nodes found in the cluster. If you have GPU nodes, please ensure that they have the label `nvidia.com/gpu: <number of GPUs>`
```

**Comment by creatorrr** [comment]
This workaround seems to work for enabling GPUs on `kind`:
https://github.com/kubernetes-sigs/kind/pull/3257#issuecomment-1607287275

**Comment by concretevitamin** [comment]
@creatorrr There's an in-review PR #2890. What GPUs do you plan to use locally?

**Comment by creatorrr** [comment]
Ah thanks for pointing the PR out @concretevitamin . Can I drop my comments there?

**Comment by Michaelvll** [comment]
This should be resolved by #2890
